### PR TITLE
Dogfooding YAML

### PR DESCRIPTION
**Pull request by aryx**
Creating a few inception examples :)

test plan:
$ semgrep-core -test_rules tests/OTHER/rules
$ make test

### CODE REVIEW

File: semgrep-core/tests/OTHER/rules/inception.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+rules:
+- id: double-id
+  message: |
+    you use multiple times the same id
+  severity: ERROR
+  languages: [yaml]
+  pattern: |
+    ...
```
Comment (aryx) [inline semgrep-core/tests/OTHER/rules/inception.yaml:8]: This is not parsed correctly currently @emjin

File: semgrep-core/tests/OTHER/rules/inception2.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+rules:
+#ERROR: double-id!
+- id: double-id
+  message: |
+    you use multiple times the same id
+  severity: ERROR
+  languages: [json, yaml]
+  patterns:
+    - pattern-inside: |
+       [{id: $X, ...}, ..., {id: $X, ...}, ...]
```
Comment (aryx) [inline semgrep-core/tests/OTHER/rules/inception2.yaml:10]: I'm able to do it by using JSON to specify the pattern, but the yaml pattern should parse the same.

File: semgrep-core/tests/yaml/misc_double_id.sgrep [hunk 3]
```diff
@@ -0,0 +1,7 @@
+...
```
Comment (aryx) [inline semgrep-core/tests/yaml/misc_double_id.sgrep:1]: @emjin this is not parsed correctly :(

File: semgrep-core/tests/yaml/yaml_rules/kubernetes_semgrep.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+rules:
+- id: kubernetes.best-practice.no-fractional-cpu-limits
+  patterns:
+  - pattern-inside: |
+      limits: ...
```
Comment (aryx) [inline semgrep-core/tests/yaml/yaml_rules/kubernetes_semgrep.yaml:5]: This another example that does not work @emjin . It works with spacegrep (see the kubernetes_spacegrep.yaml file in the same directory) but not semgrep.
Looks more complicated to fix though, nothing obviously wrong, the AST seemed ok.

**Review by emjin — APPROVED** [review]

**Pull request merged**